metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2114,6 +2117,15 @@ def parse_custom_sql(
             schema_aware=not self.config.sql_parsing_disable_schema_awareness,
         )
 
+        if parsed_result is None or parsed_result.debug_info.error:
+            message = f"Failed to extract column level lineage from datasource {datasource_urn}"
+            if parsed_result is not None and parsed_result.debug_info.error:
```

Comment:
this check is redundant, right?

Reply:
as per my understanding, not redundant

I just rewrote code below and moved it into the `parse_custom_sql` function, as there were two callers of the function doing about the same:

```
        if parsed_result is None:
            logger.info(
                f"Failed to extract column level lineage from datasource {datasource_urn}"
            )
            return []
        if parsed_result.debug_info.error:
            logger.info(
                f"Failed to extract column level lineage from datasource {datasource_urn}: {parsed_result.debug_info.error}"
            )
        if parsed_result is None or parsed_result.debug_info.error:
            return []
```

Reply:
so `create_lineage_sql_parsed_result` will never return None - if there was an error, it would be part of the SqlParsingResult object

in debug_info, `error` is an alias for `.table_error or .column_error` (since only one of those will be set). 

https://github.com/datahub-project/datahub/blob/1bfd4ee1d47d344e1bd2970e9b214a6b3d77f41f/metadata-ingestion/src/datahub/sql_parsing/sqlglot_lineage.py#L156-L158

if there's a column error, we still can generate table-level lineage - so I think the error messages might need to be tweaked

Reply:
> so `create_lineage_sql_parsed_result` will never return None

You are right. Then definitely `parsed_result is None` becomes unnecessary. 

I made warning messages more detailed depending on the error type with commit e585d149a6b759ffb6bfc15030dfd12f5c0c165f .